<?xml version="1.0"?>

<launch>
  <arg name="gpio_name" default="anvil_misc"/>
  <arg name="phase" default="0.0"/>
  <arg name="frame_rate" default="10.0"/>
  <arg name="pulse_width_ms" default="5"/>
  <arg name="gpio_mapping_file" default="$(find sensor_trigger)/config/gpio_mapping.yaml"/>

  <node pkg="sensor_trigger" type="sensor_trigger" name="sensor_trigger" launch-prefix="taskset -c 0">
    <param name="gpio_name" value="$(var gpio_name)"/>
    <param name="phase" value="$(var phase)"/>
    <param name="frame_rate" value="$(var frame_rate)"/>
    <param name="pulse_width_ms" value="$(var pulse_width_ms)"/>
    <param name="gpio_mapping_file" value="$(var gpio_mapping_file)"/>
  </node>
</launch>
